docker-compose.yaml

```diff
@@ -13,19 +13,22 @@ services:
             - DD_APM_NON_LOCAL_TRAFFIC=true
             - DD_API_KEY=${DD_API_KEY}
             - DD_SITE=${DD_SITE:-datadoghq.com}
-            - DD_ENV=prod
-    default:
+            - DD_ENV=dev
+    hash-honeypot:
         build:
             context: .
             dockerfile: Dockerfile
+        volumes:
+            - .:/usr/src/app
         environment:
             - DD_AGENT_HOST=agent
             - DD_TRACE_AGENT_PORT=8126
             - DD_TRACER_ENABLED=true
             - DD_APPSEC_ENABLED=true
+            - DD_API_KEY=${DD_API_KEY}
             - DD_SERVICE=${DD_SERVICE:-sec-research}
             - DD_TRACE_DEBUG=false
-            - DD_ENV=prod
+            - DD_ENV=dev
```

Comment:
Nitpick: These could be expressed as Dockefile ARGS, this would also allow control in the Dockerfile is they are not supplied.